packages/@aws-cdk/core/lib/runtime.ts

```diff
@@ -55,6 +59,110 @@ export function cfnTagToCloudFormation(x: any): any {
   };
 }
 
+export function cloudFormationTagToCdkTag(tag: any): CfnTag {
+  return tag == null
+    ? { } as any // break the type system - this should be detected at runtime by a tag validator
+    : {
+      key: tag.Key,
+      value: tag.Value,
+    };
+}
+
+// nothing to do - if 'value' is not a boolean or a Token,
+// a validator should report that at runtime
+export const booleanFromCloudFormation: (value: any) => boolean | IResolvable = identity;
+
+// nothing to for any but return it
+export const anyFromCloudFormation = identity;
+
+export function dateFromCloudFormation(value: any): Date | IResolvable {
```

Comment:
Dates are silly. Don't think we actually use them anywhere.

Reply:
We definitely do (for example, it's the type of `CfnBucket.RuleProperty.expirationDate` in S3).